Rename nbde_server_create_keys to nbde_server_rotate_keys

sergio-correia opened the pull request:
The behavior is the following:

    - if there are no keys, create them
    - if there are keys, do not touch them, unless nbde_server_create_keys is
      set to `yes', in which case the keys will be rotated and new keys will
      be created

sergio-correia commented:
@richm: I have rebased this PR on top of master, but how can I be sure ` tests/test_nbde_server_create_keys.yml` is running in the CI?

richm commented:
> @richm: I have rebased this PR on top of master, but how can I be sure ` tests/test_nbde_server_create_keys.yml` is running in the CI?

Looks like nbde_server is not enabled in CI. I'll have to enable it.

Also, the test file should be called `tests/tests_nbde_server_create_keys.yml` - see https://github.com/linux-system-roles/test-harness/blob/master/test/run-tests#L373

richm commented:
don't worry about the el6 failures - nbde isn't supported on el6, but there isn't currently a way for a project to specify that for CI

sergio-correia commented:
> > @richm: I have rebased this PR on top of master, but how can I be sure ` tests/test_nbde_server_create_keys.yml` is running in the CI?
> 
> Looks like nbde_server is not enabled in CI. I'll have to enable it.
> 
> Also, the test file should be called `tests/tests_nbde_server_create_keys.yml` - see https://github.com/linux-system-roles/test-harness/blob/master/test/run-tests#L373

Thanks, I have fixed the test name.

richm commented on tests/tests_nbde_server_rotate_keys.yml:
```diff
@@ -0,0 +1,67 @@
+
+    - name: Check whether keys changed - nbde_server_rotate_keys not set
+      assert:
+        that: "{{ existing_keys.files == existing_keys_rotate_not_set.files }}"
```
Is it necessary to have the test enclosed in `"{{ .... }}"`?  If it is not needed, I would prefer to drop it.
I know there are other places in the code where there are `when` and `that` tests that use `"` - don't worry about those places which are not directly related to this PR, but please change this and other `that` tests to not enclose the tests in `"{{ .... }}"` if they are not needed.

sergio-correia replied:
I tested locally and it seems we don't need those in these cases, so I will be removing them.

richm commented on tests/tests_nbde_server_rotate_keys.yml:
```diff
@@ -0,0 +1,67 @@
+
+    - name: Check whether keys changed - nbde_server_rotate_keys set
+      assert:
+        that: "{{ existing_keys.files != existing_keys_rotate_set.files }}"
```
here

richm commented on tests/verify-role-results.yml:
```diff
@@ -0,0 +1,44 @@
+
+- name: Check whether required packages are installed
+  assert:
+    that: "{{ not packages_installed.changed }}"
```
here

richm commented on tests/verify-role-results.yml:
```diff
@@ -0,0 +1,44 @@
+- name: Check whether service is enabled and started when it should be
+  when: state is not defined or state != "stopped"
+  assert:
+    that: "{{ not nbde_service_started.changed }}"
```
here

richm commented on tests/verify-role-results.yml:
```diff
@@ -0,0 +1,44 @@
+- name: Check whether service is enabled and stopped when it should be
+  when: state is defined or state == "stopped"
+  assert:
+    that: "{{ not nbde_service_stopped.changed }}"
```
here

richm commented on tests/verify-tang-results.yml:
```diff
@@ -0,0 +1,14 @@
+
+- name: Check whether there are keys
+  assert:
+    that: "{{ not existing_keys.changed }}"
```
here

richm commented on tests/tests_default_vars.yml:
```diff
@@ -7,7 +7,7 @@
             name: nbde_server
         - assert:
-            that: "{{ item }} is defined"
+            that: item is defined
```
I think `"{{ item }} is defined"` is correct.  I tried a simple playbook like this:
```yaml
- name: set parameters for test
  hosts: localhost
  tasks:
  - set_fact:
      astring: "astringval"
  - set_fact:
      bstring: "bstringval"
  - debug:
      msg: "{{ item }} is defined"
    loop:
      - astring
      - bstring
      - cstring
  - assert:
      that: item is defined
    loop:
      - astring
      - bstring
      - cstring
```
This incorrectly reported that cstring is defined.
```
ok: [localhost] => (item=cstring) => {
    "ansible_loop_var": "item",
    "changed": false,
    "item": "cstring",
    "msg": "All assertions passed"
}
```

sergio-correia replied:
Heh, I did this one last, and it seemed fishy, but passed the test. Let me revert this portion.
